platforms/jvm/toolchains-jvm/build.gradle.kts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("gradlebuild.distribution.api-java")
+    id("gradlebuild.distribution.api-kotlin")
+}
+
+description = "Adds support for using JVM toolchains in projects"
+
+dependencies {
+    implementation(project(":base-services"))
+    implementation(project(":core"))
+    implementation(project(":core-api"))
+    implementation(project(":dependency-management"))
+    implementation(project(":diagnostics"))
+    implementation(project(":enterprise-operations"))
+    implementation(project(":file-collections"))
+    implementation(project(":jvm-services"))
+    implementation(project(":logging"))
+    implementation(project(":model-core"))
+    implementation(project(":persistent-cache"))
+    implementation(project(":platform-base"))
+    implementation(project(":platform-jvm"))
+    implementation(project(":resources"))
+
+    implementation(libs.commonsIo)
+    implementation(libs.commonsLang)
+    implementation(libs.groovy)
+    implementation(libs.guava)
+    implementation(libs.inject)
+    implementation(libs.nativePlatform) {
+        because("Required for SystemInfo")
+    }
+    implementation(libs.futureKotlin("stdlib-jdk8"))
+
+    testImplementation(testFixtures(project(":core")))
+    testImplementation(testFixtures(project(":language-groovy")))
+    testImplementation(testFixtures(project(":logging")))
+
+    testRuntimeOnly(project(":distributions-core")) {
+        because("Tests instantiate DefaultClassLoaderRegistry which requires a 'gradle-plugins.properties' through DefaultPluginModuleRegistry")
+    }
+
+    integTestImplementation(testFixtures(project(":model-core")))
+    integTestImplementation(testFixtures(project(":language-groovy")))
```

Comment:
After some experimentation, here's what I think.

These fixtures should properly live in the `language-groovy` project, as that's where the `GroovyCompiler` types live.

`language-groovy` properly deps `toolchains-jvm` as an implementation dependency.

`plugins` deps the test fixtures of `language-groovy` so that integration tests can extend the base types contained in these fixtures.

Currently, `toolchains-jvm` also deps the test fixtures of `language-groovy`.  This isn't a problem for the build, but is kind of odd to see such a circular chain of dependencies.  This is due to the integration tests in the `toolchains-jvm` project testing groovy compilation scenarios and needing to extend the base types contained in these fixtures.

We could:

1. Move the extending integration tests to a new or existing project that deps both `toolchains-jvm` and `language-groovy`.  This would allow removing the `toolchains-jvm` dep on the test fixtures.  `integ-test` would work, but that's really just postponing dealing with the problem, since we want to divide that project into platforms and remove it.
2. Move the fixtures into a new or existing project that is depped by both `toolchains-jvm` and `language-groovy`.  Perhaps `platform-jvm`?  I don't like that this moves the fixtures away from the Groovy compiler types they are meant to test.
3. Move the tests out of `plugins` into `toolchains-jvm`.  Move the test fixtures from `language-groovy` into `toolchains-jvm`.  Now there is no need for anything to dep the `language-groovy` test fixtures, since they won't exist - everything will be located inside the `toolchains-jvm` project.  I think this is less correct, and bloats that project.
4. Do nothing
5. Do something else

Unless you have a good option 5, I favor option 4 as it seems to leave the classes where you would expect to find them.  I can add a `because` in the `toolchains-jvm` project's dep on the `language-groovy` test fixtures to lampshade the unusual dep cycle, but that cycle isn't hurting anything ATM.  Sterling and I did discuss test fixture cycles briefly and he was okay with it and approved this PR.  All else equal it would be better not to have it, but I think co-locating the tests and fixtures with the correct prod code is more important.  If this becomes a problem later it can be addressed then.